### ISSUE TITLE
azure.mgmt.eventhub.models.EHNamespace.disable_local_auth not recognized by mypy

### ISSUE THREAD

**Issue by jazelenk** [Azure/azure-sdk-for-python#37901]
- **Package Name**: azure.mgmt.eventhub
- **Package Version**: 11.1.0
- **Operating System**: Linux
- **Python Version**: 3.12

**Describe the bug**
mypy claims that azure.mgmt.eventhub.models.EHNamespace has no disable_local_auth attribute

**To Reproduce**
$ cat example.py
import azure.mgmt.eventhub.models

ehn = azure.mgmt.eventhub.models.EHNamespace()
print(f"ehn.disable_local_auth is {ehn.disable_local_auth!r}")

$ python3 example.py
ehn.disable_local_auth is None

$ mypy example.py
example.py:4:35: error: "EHNamespace" has no attribute "disable_local_auth"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)

 

**Expected behavior**
I expect mypy to recognize that disable_local_auth is a valid attribute

**Screenshots**
see repro above

**Additional context**
n/a

**Comment by kashifkhan** [comment]
Thank you for the feedback @jazelenk . We will investigate this and get back to you asap.

**Comment by kashifkhan** [comment]
@msyyc can you assist on this one please ?

**Comment by msyyc** [comment]
Let me take an investigation.

**Comment by msyyc** [comment]
It may be related with package structure. "azure-mgmt-eventhub" is multi-api packge which contains many api-version folder named `v20XX_XX_XX`. In old api-version folder (e.g. [v2017_04_01](https://github.com/Azure/azure-sdk-for-python/blob/dc9085905d6d6d5e5c8eb33ff82073420c4f0312/sdk/eventhub/azure-mgmt-eventhub/azure/mgmt/eventhub/v2017_04_01/models/_models_py3.py#L708-L717)), the model has no attribute `disable_local_auth`. While in latest api-version (e.g. [v2024_04_01](https://github.com/Azure/azure-sdk-for-python/blob/dc9085905d6d6d5e5c8eb33ff82073420c4f0312/sdk/eventhub/azure-mgmt-eventhub/azure/mgmt/eventhub/v2024_01_01/models/_models_py3.py#L1339)), the model has `disable_local_auth`. However, mypy analyze this model with old api-version instead of latest api-version which results in the fake error.

**Comment by jazelenk** [comment]
Yeah - the multiple imports in the generated models.py looks funky. That ends up redefining a bunch of stuff.

You can see this if you reorder the importing - if the newer v2024 comes before the older v2017, everything breaks.

It'd be slick if autorest recognized that newer versions redefine/augment the same datatypes and avoided generating redundant definitions. It's also a lot less work to say that than it is to implement. :-)

**Comment by msyyc** [comment]
We will drop multiapi support for `azure-mgmt-eventhub` in short future then the issue shall be fixed.
